tracer/src/Datadog.Trace/ContinuousProfiler/ProfilerAvailabilityHelper.cs

```diff
@@ -0,0 +1,58 @@
+#nullable enable
+
+using System;
+using Datadog.Trace.Util;
+
+namespace Datadog.Trace.ContinuousProfiler;
+
+internal static class ProfilerAvailabilityHelper
+{
+    // This will never change, so we use a lazy to cache the result.
+    // This confirms that we are in an automatic instrumentation environment (and so P/Invokes have been re-written)
+    // and that the profiling library has been deployed (which is not the case in some serverless environments).
+    // We should add or remove conditions from here as our deployment requirements change.
+    // Longer term, we'd like to be able to pass this information from the native side to the managed side, but
+    // today that only works on Windows (hence the early short-circuit).
+    private static readonly Lazy<bool> ProfilerIsAvailable = new(() =>
+    {
+        // Profiler is not available on ARM(64)
+        var fd = FrameworkDescription.Instance;
+        if (!IsSupportedArch(fd))
+        {
+            return false;
+        }
+
+        // This variable is set by the native loader after trying to load the profiler
+        // it means the profiler is _there_ though it may not be _loaded_. This only works
+        // on Windows at the moment. We assume that the CLR profiler must be attached in this scenario
+        if (fd.IsWindows() && !string.IsNullOrEmpty(EnvironmentHelpers.GetEnvironmentVariable("DD_INTERNAL_PROFILING_NATIVE_ENGINE_PATH")))
+        {
+            return true;
+        }
+
+        // Now we're into fuzzy territory. The CP is not available in some environments
+        // - AWS Lambda
+        // - Azure Functions where the site extension is _not_ used
+        var isUnsupported = EnvironmentHelpers.IsAwsLambda()
```

Comment:
is it ok if it tries logging here? as this ends up [there](https://github.com/DataDog/dd-trace-dotnet/blob/d0d795bd1aa817e41d9ce555a2747c1eb3d5f642/tracer/src/Datadog.Trace/Util/EnvironmentHelpers.cs#L67)

 otherwise there's this [beautiful solution](https://github.com/DataDog/dd-trace-dotnet/blob/303c4f9d3dc4e4ac35ccd1094a7f7e254663ebce/tracer/src/Datadog.Trace/Util/EnvironmentHelpersNoLogging.cs#L41) 😅